state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -3038,32 +3093,49 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		T::FeePayment::try_burn_fee(requester, channel_opening_fee)?;
 		Self::deposit_event(Event::<T, I>::ChannelOpeningFeePaid { fee: channel_opening_fee });
 
-		let (deposit_channel, channel_id) = if let Some((channel_id, mut deposit_channel)) =
-			DepositChannelPool::<T, I>::drain().next()
+		// Allocate a channel for the requester, always try to first allocate from the
+		// pre-allocated channels list
+		let mut deposit_channel =
+			match PreallocatedChannels::<T, I>::mutate(requester, |queue| queue.pop_front()) {
+				Some(channel) => channel,
+				// if there are no pre-allocated channels, take the next channel
+				// from the pool of recycled channels, or generate a new channel entirely.
+				None => match DepositChannelPool::<T, I>::drain().next() {
+					Some((_, deposit_channel)) => deposit_channel,
+					None => Self::generate_new_channel(source_asset)?,
+				},
+			};
+		// Make sure to set the asset on the channel, in case it was recycled
+		deposit_channel.asset = source_asset;
```

Comment:
We should move this above into the branch above where we get the recycled channel - given it's only relevant for that case now... and makes it clear, removing the need for the comment.